### ISSUE TITLE
Move fetch into utils

### ISSUE THREAD

**Issue by jeluard** [ChainSafe/lodestar#6505]
Migrate [fetch](https://github.com/ChainSafe/lodestar/blob/unstable/packages/api/src/utils/client/fetch.ts) into the [utils](https://github.com/ChainSafe/lodestar/tree/unstable/packages/utils) package.
All references should be updated accordingly.

[tests](https://github.com/ChainSafe/lodestar/blob/unstable/packages/api/test/unit/client/fetch.test.ts) will have to be migrated too. Consider that tests requiring a web browser won't be able to run in browser environments.

**Comment by wemeetagain** [comment]
@nflaig 
Is this still a good idea? is the idea that we would use the wrapped fetch across the whole monorepo?

**Comment by nflaig** [comment]
> is the idea that we would use the wrapped fetch across the whole monorepo?

we already do, there is a biomjs rule to prevent using the global directly, I am not sure what was the reason to move it to utils package I guess related to https://github.com/ChainSafe/lodestar/issues/6488 that we wanna remove `got` and utils is a dependency of basically all packages while the api package is not.

**Comment by matthewkeil** [comment]
> [@nflaig](https://github.com/nflaig) Is this still a good idea? is the idea that we would use the wrapped fetch across the whole monorepo?

Yes.  Its a great first issue that will remove a large dependency



> we already do, there is a biomjs rule to prevent using the global directly, I am not sure what was the reason to move it to utils package I guess related to [#6488](https://github.com/ChainSafe/lodestar/issues/6488) that we wanna remove `got` and utils is a dependency of basically all packages while the api package is not.

Yes.  There was a package circular dependency.  It should be moved to a package with no other deps so its a clean import everywhere

**Comment by nflaig** [comment]
resolved

https://github.com/ChainSafe/lodestar/blob/9fa839a030f88447bee7204f87aa64a08b74e255/packages/utils/src/fetch.ts#L10
